### PR TITLE
chore(core): Flatten out core-common and buffers

### PR DESCRIPTION
**Pull request by bruceg**
Does this make sense to simplify our internal structure? I was a _little_ annoyed at the the double nesting of some of the bits while working on some core structures.

### CODE REVIEW

**Comment by tobz** [comment]
Just about to review this, but at a high level: this change sounds great to me. I've also never really been a huge fan of the nesting under `vector-core`.

**Review by tobz — APPROVED** [review]
Yeah, honestly, LGTM. 👍🏻

**Review by blt — APPROVED** [review]
I like it! The nesting is something I did because of dependency issues when we were migrating into core. Glad it can be un-nested.

File: lib/vector-buffers/Cargo.toml [hunk 1]
```diff
@@ -1,53 +1,53 @@
...
 tokio-util = { version = "0.6", default-features = false }
+tokio = { version = "1.15.0", default-features = false, features = ["rt", "macros", "rt-multi-thread", "sync", "fs", "io-util", "time"] }
 tracing = { version = "0.1.29", default-features = false, features = ["attributes"] }
+vector_common = { path = "../vector-common", default-features = false }
 
 [dev-dependencies]
 clap = "3.0.10"
 criterion = { version = "0.3", features = ["html_reports", "async_tokio"] }
 hdrhistogram = "7.3.0"
+# https://github.com/tobz/tracing-fluent-assertions/pull/1
```
[the remark this answers is not in the document]
Comment (bruceg) [reply lib/vector-buffers/Cargo.toml]: Fixed in f3f4e2328

**Review by lukesteensen — APPROVED** [review]

**Pull request merged**